social-network/src/YourArticles.js

```diff
@@ -0,0 +1,48 @@
+// yourArticles.js
+import React from 'react';
+import yourArticlesJson from './your-articles.json';
+
+class yourArticlesComp extends React.Component {
+  
+    render() {
+        const articles = [];
+        for (var i=0; i < yourArticlesJson.length; i++) {
+          articles.push(yourArticlesJson[i]);
+        }
+        const articlesMod = articles.map(array => { 
+          array = { ...array, postedDate: array.postedDate.substring(0, 10) };
+          return array;
+        });
+
+        function Articles(props) {  
+          const headline = <h4 class='Articles-headline'>For you</h4>;
+          const content = props.articles.map((articles) =>
+            <div key={articles.title} class='YourArticles-content'>
```

Comment:
Note that the code here is essentially the same as the code in `MissedArticles`. This code is a great candidate for a separate component, maybe `Article`. While the classes differ, you can define different classnames for the component depending on where they're rendered. e.g. in YourArticles:

```js
props.articles.map(article => <Article data={article} className="YourArticle" />)
```

In MissedArticles:

```js
```js
props.articles.map(article => <Article data={article} className="MissedArticle" />)
```
```